Tiletactics.java

```diff
@@ -0,0 +1,349 @@
+            case 16 -> Constants.COLOR_RED + value + Constants.COLOR_RESET;
+            case 32 -> Constants.COLOR_PURPLE + value + Constants.COLOR_RESET;
+            case 64 -> Constants.COLOR_CYAN + value + Constants.COLOR_RESET;
+            case 128 -> Constants.COLOR_LIGHT_YELLOW + value + Constants.COLOR_RESET;
+            case 256 -> Constants.COLOR_LIGHT_PURPLE + value + Constants.COLOR_RESET;
+            case 512 -> Constants.COLOR_LIGHT_CYAN + value + Constants.COLOR_RESET;
+            case 1024 -> Constants.COLOR_LIGHT_RED+ value + Constants.COLOR_RESET;
+            case 2048 -> Constants.COLOR_LIGHT_GREEN + value + Constants.COLOR_RESET;
+            default -> value == 0 ? "" : Integer.toString(value);
+        };
+    }
+
+    // Add a random tile (either 2 or 4) to an empty spot on the board
+    // output - Updates the board with a random tile
+    public void addRandomTile() {
+        int rowIndex, columnIndex;
+        do {
+            rowIndex = randomGenerator.nextInt(SIZE);
+            columnIndex = randomGenerator.nextInt(SIZE);
+        } while (board[rowIndex][columnIndex] != 0);
+        board[rowIndex][columnIndex] = (randomGenerator.nextInt(10) < 9) ? 2 : 4;
+    }
+
+    // Shift tiles left and merge if possible
+    // output - Moves and merges tiles to the left
+    public void moveLeft() {
+        for (int row = 0; row < SIZE; row++) {
+            int[] compressedRow = new int[SIZE];
+            int insertPosition = 0;
+
+            for (int column = 0; column < SIZE; column++) {
+                if (board[row][column] != 0) {
+                    compressedRow[insertPosition++] = board[row][column];
+                }
+            }
+
+            for (int i = 0; i < insertPosition - 1; i++) {
+                if (compressedRow[i] == compressedRow[i + 1]) {
+                    int mergedValue = compressedRow[i] * 2;
+                    compressedRow[i] = mergedValue;
+                    compressedRow[i + 1] = 0;
+                    score += mergedValue;
+                    if (mergedValue == 2048)
+                        gameWon = true;
+                    i++;
+                }
+            }
+
+            int[] newRow = new int[SIZE];
+            int newIndex = 0;
+            for (int value : compressedRow) {
+                if (value != 0) {
+                    newRow[newIndex++] = value;
+                }
+            }
+            board[row] = newRow;
+        }
+    }
+
+    // Shift tiles right and merge if possible
+    // output - Moves and merges tiles to the right
+    public void moveRight() {
+        for (int row = 0; row < SIZE; row++) {
+            int[] compressedRow = new int[SIZE];
+            int insertPosition = SIZE - 1;
+
+            for (int column = SIZE - 1; column >= 0; column--) {
+                if (board[row][column] != 0) {
+                    compressedRow[insertPosition--] = board[row][column];

```

Comment:
don't perform increment and decrement operation directly while referring it, inc/dec it before and then use